operators/src/processing/reprojection.rs

```diff
@@ -0,0 +1,399 @@
+    fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<InitializedVectorOperator>> {
+        ensure!(
+            self.vector_sources.len() == 1,
+            error::InvalidNumberOfVectorInputs {
+                expected: 1..2,
+                found: self.vector_sources.len()
+            }
+        );
+        ensure!(
+            self.raster_sources.is_empty(),
+            error::InvalidNumberOfRasterInputs {
+                expected: 0..1,
+                found: self.raster_sources.len()
+            }
+        );
+
+        let initialized_vector_sources = self
+            .vector_sources
+            .into_iter()
+            .map(|o| o.initialize(context))
+            .collect::<Result<Vec<Box<InitializedVectorOperator>>>>()?;
+
+        let in_desc: &VectorResultDescriptor = initialized_vector_sources[0].result_descriptor();
+        let out_desc = VectorResultDescriptor {
+            spatial_reference: self.params.target_spatial_reference.into(),
+            data_type: in_desc.data_type,
+            columns: in_desc.columns.clone(),
+        };
+
+        let state = (
+            Option::from(in_desc.spatial_reference).unwrap(),
+            self.params.target_spatial_reference,
+        );
+
+        Ok(
+            InitializedReprojection::new((), out_desc, vec![], initialized_vector_sources, state)
+                .boxed(),
+        )
+    }
+}
+
+impl InitializedOperator<VectorResultDescriptor, TypedVectorQueryProcessor>
+    for InitializedReprojection
+{
+    fn query_processor(&self) -> Result<TypedVectorQueryProcessor> {
+        let state = self.state;
+        match self.vector_sources[0].query_processor()? {
+            // TODO: use macro for that
+            TypedVectorQueryProcessor::Data(source) => Ok(TypedVectorQueryProcessor::Data(
+                MapQueryProcessor::new(source, move |query| {
+                    query_rewrite_fn(query, state.0, state.1)
+                })
+                .boxed(),
+            )),
+            TypedVectorQueryProcessor::MultiPoint(source) => {
+                Ok(TypedVectorQueryProcessor::MultiPoint(
+                    ReprojectionProcessor::new(source, self.state.0, self.state.1).boxed(),
+                ))
+            }
+            TypedVectorQueryProcessor::MultiLineString(source) => {
+                Ok(TypedVectorQueryProcessor::MultiLineString(
+                    ReprojectionProcessor::new(source, self.state.0, self.state.1).boxed(),
+                ))
+            }
+            TypedVectorQueryProcessor::MultiPolygon(source) => {
+                Ok(TypedVectorQueryProcessor::MultiPolygon(
+                    ReprojectionProcessor::new(source, self.state.0, self.state.1).boxed(),
+                ))
+            }
+        }
+    }
+}
+
+struct ReprojectionProcessor<Q, G> {
+    source: Q,
+    from: SpatialReference,
+    to: SpatialReference,
+    geom_type: PhantomData<G>,
```

Comment:
I guess you don't need the `geom_type` if you say source is a `VectorQueryProcessor<VectorType = FeatureCollection<G>>,`

Reply:
6ce903c2d897